consultation_analyser/consultations/views/answers.py

```diff
@@ -3,52 +3,274 @@
 
 from django.core.exceptions import ObjectDoesNotExist
 from django.core.paginator import Paginator
+from django.db.models import Count, F, Q, QuerySet, Sum, Value
+from django.db.models.functions import Length, Replace
 from django.http import HttpRequest
 from django.shortcuts import get_object_or_404, redirect, render
 
 from .. import models
 from .decorators import user_can_see_consultation
 
 
+def filter_by_response_and_theme(
+    question: models.Question,
+    respondents: QuerySet,
+    responseid: str | None = None,
+    responsesentiment: list[str] | None = None,
+    themesfilter: list[str] | None = None,
+    themesentiment: list[str] | None = None,
+) -> QuerySet:
+    """filter respondents by response themes"""
+    if responseid:
```

Comment:
Should this be a filter by the `themefinder_response_id` as integers are easier to deal with? Even if the label should be "Response ID".

Can be addressed in a separate PR - and potentially one to check with Jude etc.